### ISSUE TITLE
Training Dataset: also a test dataset with labels must be created to calculate metrics on new data

### ISSUE THREAD

**Issue by barbara1990** [boku-ilen/darknet#38]
Metrics generated on validation dataset are not enough, since the model use this data to perform better. A new set of labeled data is needed:
1. Split already labeled data into tree parts: training, validation, test 
2. Calculate metrics for test data

**Comment by barbara1990** [comment]
20200403: 1883 images (with needed text files) have been split into 80% for training, 10% for validation and 10% for testing and trained for 9000 iterations with batches of 64 images. The console output has been saved in .out file: e.g. training loss for each iteration, validation mAP-0.50 (and other metrics) for every 100 iterations. Weights have been saved each 1000 iterations and additionally best_weights (for 5500 iteration), which gives the best result on validation data and these weights will be used for detection on further images. The performance for weights generated every 1000 iterations is presented in the [mAP graph](https://github.com/boku-ilen/darknet/blob/master/results/20200403_mAPgraph.jpg) (values for test data). Consider continuing or improving data sets for getting even better results. 

2020401: 1883 images (with needed text files) have been split into 60% for training, 20% for validation and 20% for testing and trained for 4800 iterations with batches of 64 images. The training was too short and it had been decided to split data in other way and start a longer training again (see 20200403).
